### PR TITLE
ci: migrate Vale usage to vale-action

### PR DESCRIPTION
**Pull request by SilencerWeb**
This PR migrates Vale usage to [vale-action](https://github.com/errata-ai/vale-action)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,6 @@
     "lint:fix": "eslint --ext .js,.jsx --fix .",
     "lint:examples": "eslint -c ./src/data/markdown/.eslintrc.js src/data/markdown",
     "lint:examples:fix": "eslint -c ./src/data/markdown/.eslintrc.js --fix src/data/markdown",
-    "lint:prose": "vale --glob='**/{docs,translated-guides/en}/**/*.md' ./src/data/markdown",
```
Comment (ppcano) [inline package.json:26]: @SilencerWeb, this command is necessary to validate it locally.

Comment (SilencerWeb) [reply package.json]: @ppcano returned it back

**Comment by ppcano** [comment]
@SilencerWeb, can we merge this?

**Comment by SilencerWeb** [comment]
@ppcano yup!

**Pull request merged**